plugins/org.komodo.core/src/org/komodo/repository/ObjectImpl.java

```diff
@@ -68,6 +71,87 @@ ObjectImpl accessOuter() {
     }
 
     /**
+     * Implementation specific and not to be included on the interface!
+     *
+     * @param uow transaction for this operation. Can be null.
+     *
+     * @return the underlying object of this {@link ObjectImpl}
+     *
+     * @throws KException if an error occurs
+     */
+    public Node node(UnitOfWork uow) throws KException {
```

Comment:
Why is this method public? This is the **first** method to expose JCR to the outside world. And the comment about "not to be included on the interface" is a bit scary.